tests/topotests/bgp_flowspec/test_bgp_flowspec_topo.py

```diff
@@ -0,0 +1,170 @@
+sys.path.append(os.path.join(CWD, '../'))
+
+# pylint: disable=C0413
+# Import topogen and topotest helpers
+from lib import topotest
+from lib.topogen import Topogen, TopoRouter, get_topogen
+from lib.topolog import logger
+from lib.lutil import lUtil
+from lib.lutil import luCommand
+
+# Required to instantiate the topology builder class.
+from mininet.topo import Topo
+
+#####################################################
+##
+##   Network Topology Definition
+##
+#####################################################
+
+class BGPFLOWSPECTopo1(Topo):
+    "BGP EBGP Flowspec Topology 1"
+
+    def build(self, **_opts):
+        tgen = get_topogen(self)
+
+        # Setup Routers
+        tgen.add_router('r1')
+
+        # Setup Control Path Switch 1. r1-eth0
+        switch = tgen.add_switch('s1')
+        switch.add_link(tgen.gears['r1'])
+
+        ## Add eBGP ExaBGP neighbors
+        peer_ip = '10.0.1.101' ## peer
+        peer_route = 'via 10.0.1.1' ## router
+        peer = tgen.add_exabgp_peer('peer1',
+                                    ip=peer_ip, defaultRoute=peer_route)
+        switch.add_link(peer)
+
+#####################################################
+##
+##   Tests starting
+##
+#####################################################
+
+def setup_module(module):
+    tgen = Topogen(BGPFLOWSPECTopo1, module.__name__)
+
+    tgen.start_topology()
+    # check for zebra capability
+    router = tgen.gears['r1']
+
+    # Get r1 reference and run Daemons
+    logger.info('Launching BGP and ZEBRA on r1')
+    router = tgen.gears['r1']
+    router.load_config(
+        TopoRouter.RD_ZEBRA,
+        os.path.join(CWD, '{}/zebra.conf'.format('r1'))
+    )
+    router.load_config(
+        TopoRouter.RD_BGP,
+        os.path.join(CWD, '{}/bgpd.conf'.format('r1'))
+    )
+    router.start()
+
+    peer_list = tgen.exabgp_peers()
+    for pname, peer in peer_list.iteritems():
+        peer_dir = os.path.join(CWD, pname)
+        env_file = os.path.join(CWD, "exabgp.env")
+        peer.start(peer_dir, env_file)
+        logger.info(pname)
+
+
+def teardown_module(module):
+    tgen = get_topogen()
+    tgen.stop_topology()
+    
+def test_bgp_convergence():
+    "Test for BGP topology convergence"
+    tgen = get_topogen()
+
+    # Skip if previous fatal error condition is raised
+    topotest.sleep(10, 'starting BGP peering with peer1')
+    if tgen.routers_have_failure():
+        pytest.skip(tgen.errors)
+
+def test_bgp_flowspec():
+    tgen = get_topogen()
+
+    # Skip if previous fatal error condition is raised
+    if tgen.routers_have_failure():
+        pytest.skip(tgen.errors)
+
+    router = tgen.gears['r1']
+
+    logger.info('Check BGP FS entry for 3.3.3.3 with redirect IP')
+    output = router.vtysh_cmd('show bgp ipv4 flowspec 3.3.3.3', isjson=False, daemon='bgpd')
+    logger.info(output)
+    if 'NH 50.0.0.2' not in output or 'FS:redirect IP' not in output or 'Packet Length < 200' not in output:
```

Comment:
Please run this file with `black` formatter, we switched to it like here: http://docs.frrouting.org/projects/dev-guide/en/latest/topotests.html#writing-a-new-test ^f+"Format the new code using black"

Reply:
do you have a patch that I could apply to format the test ?
it is good to propose new tools.
the best would be that a patch should be proposed too in order to help developer.

for example, code formating with frr code is fun now.

Reply:
Afaik, Quentin is adding the same as with clang-formatter to polychaeta. But for now, nothing specific, we need just to run the `pip install black ; black tests/topotests/bgp_flowspec/test_bgp_flowspec_topo.py `.

Reply:
ok understood and pushed. thanks